src/components/Calculator.js

```diff
@@ -1,18 +1,44 @@
-/* eslint-disable react/prefer-stateless-function */
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+    };
+  }
+
+  clickAction = (e) => {
+    this.setState((state) => calculate(state, e.target.innerHTML));
+  }
+
   render() {
     const digitsArray = [7, 8, 9, 4, 5, 6, 1, 2, 3, 0, '.'];
     const utilsArray = ['AC', '+/-', '%'];
     const operatorsArray = ['/', 'x', '-', '+', '='];
+    const { total, next } = this.state;
+
+    const utils = utilsArray.map((utilLabel, index) => (
+      <button type="button" id={`util-${index}`} key={`${utilLabel}`} className="util" onClick={this.clickAction}>
+        {utilLabel}
+      </button>
+    ));
+
+    const digits = digitsArray.map((digitLabel, index) => (
+      <button type="button" key={`${digitLabel}`} id={`digit-${index}`} className="digit" onClick={this.clickAction}>
+        {digitLabel}
+      </button>
+    ));
 
-    const utils = utilsArray.map((utilLabel, index) => <div id={`util-${index}`} key={`${utilLabel}`} className="util">{utilLabel}</div>);
-    const digits = digitsArray.map((digitLabel, index) => <div key={`${digitLabel}`} id={`digit-${index}`} className="digit">{digitLabel}</div>);
-    const operators = operatorsArray.map((operatorLabel, index) => <div id={`operator-${index}`} key={`${operatorLabel}`} className="operator">{operatorLabel}</div>);
+    const operators = operatorsArray.map((operatorLabel) => (
+      <button type="button" key={`${operatorLabel}`} className="operator" onClick={this.clickAction}>
+        {operatorLabel}
+      </button>
+    ));
```

Comment:
- [ ] Your division operation is not working outputting this error: ![image](https://user-images.githubusercontent.com/38522578/134443789-22207dd0-5be7-4d8c-8a39-8a5d7b1392ba.png) please take a look at it and fix it to comply with the project requirements 👍

Reply:
Thanks for the feedback. I corrected the bug with the division operation.